pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -581,6 +584,48 @@ func (qjm *XController) GetAggregatedResourcesPerGenericItem(cqj *arbv1.AppWrapp
 
 	return retVal
 }
+
+//Gets all objects owned by AW from API server, check user supplied status and set whole AW status
+func (qjm *XController) GetAllObjectsOwned(caw *arbv1.AppWrapper) arbv1.AppWrapperState {
+
+	// Get all pods and related resources
+	countCompletedItems := 0
+	countCompletionRequired := 0
+	for _, genericItem := range caw.Spec.AggrResources.GenericItems {
+		objectName := genericItem.GenericTemplate
+		var unstruct unstructured.Unstructured
+		unstruct.Object = make(map[string]interface{})
+		var blob interface{}
+		if err := jsons.Unmarshal(objectName.Raw, &blob); err != nil {
+			klog.Errorf("Error unmarshalling, err=%#v", err)
+		}
+
+		status := qjm.genericresources.GetGenericItemKindStatus(&genericItem, caw.Namespace)
+		if status {
+			countCompletedItems = countCompletedItems + 1
+		}
+
+		//only consider count completion required for valid items
+		if len(genericItem.CompletionStatus) > 0 {
+			countCompletionRequired = countCompletionRequired + 1
+		}
+	}
+	klog.Infof("countCompletedItems %v, countCompletionRequired %v, podsRunning %v, podsPending %v", countCompletedItems, countCompletionRequired, caw.Status.Running, caw.Status.Pending)
+
+	//Set new status only when completion required flag is present in genericitems array
+	if countCompletedItems > 0 && countCompletionRequired > 0 {
```

Comment:
Here countCompletedItems can changed to a `bool`: `completionStatusFound` on detection of any item with `completionstatus`.

Note: If the logic gets to this statement all `completionstatus` have been met.